mountpoint-s3/src/prefetch.rs

```diff
@@ -20,14 +20,20 @@ use futures::stream::StreamExt;
 use futures::task::{Spawn, SpawnExt};
 use metrics::counter;
 use mountpoint_s3_client::{ETag, GetObjectError, ObjectClient, ObjectClientError};
+use mountpoint_s3_crt::checksums::crc::checksums_crc32c;
 use thiserror::Error;
 use tracing::{debug_span, error, trace, Instrument};
 
 use crate::prefetch::part::Part;
 use crate::prefetch::part_queue::PartQueue;
 use crate::sync::{Arc, RwLock};
 
-type TaskError<Client> = ObjectClientError<GetObjectError, <Client as ObjectClient>::ClientError>;
+use self::part::PartMismatchError;
+
+/// CRC32C checksum
+pub type ChecksumCrc = u32;
```

Comment:
I wonder if we should make a version of this typed interface part of the CRT bindings. It prevents us from confusing checksums with other values, or confusing checksums of different types. It would just be a `struct Crc32c(u32)` and the same for `Crc32`.

Actually, I wonder if we should implement a [`Hasher`](https://doc.rust-lang.org/std/hash/trait.Hasher.html) for the checksums in the CRT bindings too. That might be the more Rust idiomatic way to do it.

Reply:
Sounds good, I will move the type to `mountpoint-s3-crt`.

I was thinking about implementing the hasher as well, just not sure if we want to deviate from CRT's interface. But yeah, I would prefer the hasher implementation.